lib/circuitry/subscriber.rb

```diff
@@ -173,7 +174,11 @@ def try_with_lock(id)
     # http://www.mikeperham.com/2015/05/08/timeout-rubys-most-dangerous-api/
     def handle_message(message, &block)
       Timeout.timeout(timeout) do
-        block.call(message.body, message.topic.name)
+        if async_delete
+          block.call(message.body, message.topic.name, lambda { delete_message(message) })
```

Comment:
@diclophis thoughts on this lamda as a way to allow the message processing function to delete the message? Seems like it could be pretty useful/easy?